invenio_accounts_rest/views_auth.py

```diff
@@ -0,0 +1,477 @@
+            blueprint.add_url_rule(
+                '/forgot-password',
+                view_func=ForgotPasswordView.as_view('forgot_password'))
+            blueprint.add_url_rule(
+                '/reset-password',
+                view_func=ResetPasswordView.as_view('reset_password'))
+
+        if security_state.confirmable:
+            blueprint.add_url_rule(
+                '/send-confirmation-email',
+                view_func=SendConfirmationEmailView.as_view(
+                    'send_confirmation'))
+
+            blueprint.add_url_rule(
+                '/confirm-email',
+                view_func=ConfirmEmailView.as_view('confirm_email'))
+
+        # TODO: Check this
+        if app.config['ACCOUNTS_SESSION_ACTIVITY_ENABLED']:
+            blueprint.add_url_rule(
+                '/revoke-session',
+                view_func=RevokeSessionView.as_view('revoke_session'))
+    return blueprint
+
+
+
+class FlaskParser(FlaskParserBase):
+
+    # TODO: Add error codes to all messages (e.g. 'user-already-exists')
+    def handle_error(self, error, *args, **kwargs):
+        if isinstance(error, ValidationError):
+            _errors = []
+            for field, messages in error.messages.items():
+                _errors.extend([FieldError(field, msg) for msg in messages])
+            raise RESTValidationError(errors=_errors)
+        super(FlaskParser, self).handle_error(error, *args, **kwargs)
+
+
+webargs_parser = FlaskParser()
+use_args = webargs_parser.use_args
+use_kwargs = webargs_parser.use_kwargs
+
+
+#
+# Field validators
+#
+def user_exists(email):
+    """Validate that a user exists."""
+    if not current_datastore.get_user(email):
+        raise ValidationError(get_message('USER_DOES_NOT_EXIST')[0])
+
+
+def unique_user_email(email):
+    """Validate unique user email."""
+    if current_datastore.get_user(email) is not None:
+        raise ValidationError(
+            get_message('EMAIL_ALREADY_ASSOCIATED', email=email)[0])
+
+
+def default_user_payload(user):
+    return {
+        'id': user.id,
+        'email': user.email,
+        'confirmed_at': user.confirmed_at.isoformat() if user.confirmed_at else None,
+        'last_login_at': user.last_login_at.isoformat() if user.last_login_at else None,
+        # TODO: Check roles
+        'roles': [role_to_dict(role) for role in user.roles],
+    }
+
+
+def default_confirmation_link(user):
+    token = generate_confirmation_token(user)
+    # TODO: or hardcoded from config?
+    confirmation_link = url_for('.confirm_email', token=token, _external=True)
+    return confirmation_link, token
+
+
+def default_reset_password_link(user):
+    token = generate_reset_password_token(user)
+    # TODO: or hardcoded from config?
+    reset_link = url_for('.reset_password', token=token, _external=True)
+    return reset_link, token
+
+
+def _abort(message, field=None, status=None):
+    if field:
+        raise RESTValidationError(FieldError(field, message))
```

Comment:
`RESTValidationError ` expects a list of errors [line](https://github.com/inveniosoftware/invenio-rest/blob/cb2a265f9fb5b414c396fab12e0807ffe7bd1930/invenio_rest/errors.py#L57-L62), thus this should be wrapped in a `list(FieldError(field, message))`